packages/core/src/modules/routing/messages/StatusMessage.ts

```diff
@@ -0,0 +1,75 @@
+import { Expose, Transform } from 'class-transformer'
+import { Equals, IsBoolean, IsDate, IsInt, IsOptional, IsString } from 'class-validator'
+import { Verkey } from 'indy-sdk'
+
+import { AgentMessage } from '../../../agent/AgentMessage'
+import { ReturnRouteTypes } from '../../../decorators/transport/TransportDecorator'
+import { DateParser } from '../../../utils/transformers'
+
+export interface StatusMessageOptions {
+  id?: string
+  recipientKey?: string
+  messageCount: number
+  longestWaitedSeconds?: number
+  newestReceivedTime?: Date
+  oldestReceivedTime?: Date
+  totalBytes?: number
+  liveDelivery?: boolean
+}
+
+export class StatusMessage extends AgentMessage {
+  public constructor(options: StatusMessageOptions) {
+    super()
+    if (options) {
+      this.id = options.id || this.generateId()
+      this.recipientKey = options.recipientKey
+      this.messageCount = options.messageCount
+      this.longestWaitedSeconds = options.longestWaitedSeconds
+      this.newestReceivedTime = options.newestReceivedTime
+      this.oldestReceivedTime = options.oldestReceivedTime
+      this.totalBytes = options.totalBytes
+      this.liveDelivery = options.liveDelivery
+    }
+    this.setReturnRouting(ReturnRouteTypes.all)
```

Comment:
I normally agree that we don't need to set this in the constructor, I think it is valid in this use case though since in the [RFCs it says it is required in all of the messages. ](https://github.com/hyperledger/aries-rfcs/tree/main/features/0685-pickup-v2#reference)

Reply:
Fair point!